src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -763,7 +770,7 @@ enforcePointerValidity sc cc loc ss =
             addAssert c $ Crucible.SimError loc $
               Crucible.AssertFailureSimError msg ""
 
-       | (LLVMAllocSpec mut _pty alignment psz _ploc fresh, ptr) <- mems
+       | (LLVMAllocSpec mut _pty alignment psz _ploc fresh _sym_init, ptr) <- mems
```

Comment:
I think we need to fix this case; or perhaps the proper fix is elsewhere, I'm not sure exactly.

At any rate, if we assume in a specification that an allocation is backed by symbolic bytes, then that incurs a proof obligation at call sites to ensure that the affected memory can actually be read, and I don't see where that is being checked.

Reply:
This is by design. Similar to what @RyanGlScott is working on, this changes the semantics of memory reads such that there are always some bytes. This is not the same as the combination of `llvm_alloc` and `llvm_points_to_array_prefix`. I will update the description to make it more clear.

Reply:
That doesn't seem right.  This isn't a global option, it's a per-spec and per-allocation.  I can use `llvm_alloc` in some places and `llvm_alloc_sym_init` in some other places, right?

In that case, `llvm_alloc_sym_init` incurs an obligation we have to check for, because it isn't universal.